Automattic/woocommerce-payments
# Disable refund button when transaction disputed

Fixes https://github.com/Automattic/woocommerce-payments/issues/6378

#### Changes proposed in this Pull Request

- Disable refund button when there is active or lost dispute.
- Show explanation of to why refund button is disabled:

![Untitled](https://github.com/Automattic/woocommerce-payments/assets/73803630/9ce2cb20-fc69-4203-8beb-b9cd02ddc91b)

I refactored `DisputeNotice` a bit so it can show the explanation when refund button is disabled.
Follow the commits in this PR to understand the diff better.

When dispute is closed (status `warning_closed`), [per Stripe's docs](https://stripe.com/docs/disputes/how-disputes-work#:~:text=If%20an%20inquiry%20has%20been%20open%20for%20120%20days%20without%20escalation%20to%20a%20chargeback%2C%20Stripe%20marks%20it%20as%20closed%20in%20the%20Dashboard%20and%20API.%20At%20this%20point%2C%20you%20can%20be%20confident%20the%20card%20network%20won%E2%80%99t%20escalate%20it%E2%80%94they%20don%E2%80%99t%20provide%20an%20explicit%20%E2%80%9Cwin%E2%80%9D%20message%20for%20inquiries.), looks like refund should be allowed although I cannot test this for real. I do not disable the refund button in such case. Worst case, if Stripe does not allow refund, user will see the error.

![Screenshot 2023-07-27 at 09 14 41](https://github.com/Automattic/woocommerce-payments/assets/73803630/45cc330d-e287-4b33-9d6d-f6ca0a28e404)

#### Testing instructions

Either create transaction with different state of disputes or alter the dispute object [here](https://github.com/Automattic/woocommerce-payments/blob/5be7cb9e3485cf59915a3a562dc9305e51c5a63a/client/order/index.js#L138). I tested by altering the dispute object [here](https://github.com/Automattic/woocommerce-payments/blob/5be7cb9e3485cf59915a3a562dc9305e51c5a63a/client/order/index.js#L138), eg:
```
dispute.status = 'won';
```

- When there is no dispute:
Dispute notice should not show. Refund button should not be disabled.

- When there is a dispute needing a response:
Warning text and CTA button should show. Refund button should be disabled and there should be the explanation of to why the refund button is disabled.

![Screenshot 2023-08-23 at 01 57 23](https://github.com/Automattic/woocommerce-payments/assets/73803630/6fa11fea-57dc-41e1-875f-c14b83211cce)

- When there is a dispute that does not need a response, eg when its status is `under_review`, or when due date has passed:
Warning text and CTA button should not show. Refund button should be disabled and there should be the explanation of to why the refund button is disabled.

![Screenshot 2023-08-23 at 02 00 46](https://github.com/Automattic/woocommerce-payments/assets/73803630/baf682aa-bfd7-4beb-8ef0-ea8b071aeb62)

- When there is a dispute won / closed (status `warning_closed`):
Warning text and CTA button should not show. Refund button should not be disabled.
`ps:` you can submit a winning evidence by putting `winning_evidence` under the 'Additional details' field ([ref](https://stripe.com/docs/testing#evidence)) of the dispute challenge form.

![Screenshot 2023-08-23 at 02 22 17](https://github.com/Automattic/woocommerce-payments/assets/73803630/a93f2012-fe05-4d12-badc-8ab091419014)


-------------------


**Post merge**
